client/server/pages/analytics.js

```diff
@@ -18,14 +19,21 @@ const logAnalyticsThrottled = throttle( function ( sectionName, duration, target
  * Only logs if the request context contains a `sectionName` attribute.
  */
 export function logSectionResponse( req, res, next ) {
-	const startRenderTime = new Date();
+	const startRenderTime = process.hrtime.bigint();
 
-	res.on( 'finish', function () {
-		const context = req.context || {};
-		if ( context.sectionName ) {
-			const duration = new Date() - startRenderTime;
-			logAnalyticsThrottled( context.sectionName, duration, context.target );
+	res.on( 'close', () => {
```

Comment:
What are the "stale" requests that the `close` handler is supposed to catch, as opposed to `finished`?

Reply:
Basically any request which times out (e.g. lasts more than 50 seconds). https://logstash.a8c.com/logstash/goto/6c471944afa0cda993db44f553a56d0a

You can see in the Calypso logs that some requests are "closed" rather than "finished." These types of requests spike significantly during the outages we've had these year, but the root cause is undetermined. Some of my reading about it indicates a request might be "closed" if the network connection goes stale, but I'm not sure about that.

I'm also not sure if it's worth including in the response time metrics, because it might be a different type of scenario than "how long is the server taking to do this". At the same time, it is technically part of the overall response time

Reply:
But close also catches `finished` requests. I can't say I know everything about the internals here, but this change is partly motivated by https://github.com/Automattic/wp-calypso/pull/46937

Reply:
I see, we use the `close` event also for logging the request and saving profiles, so I guess it's consistent to use it for timings, too.